compose.prod.yml

```
@@ -80,14 +80,14 @@ services:
       - ENV=prod
       - IS_RQ=no
       - WEBPACK_MODE=production
-      - DJANGO_SETTINGS_MODULE=bagitobjecttransfer.settings.docker_prod
+      - DJANGO_SETTINGS_MODULE=app.settings.docker_prod
     depends_on:
       - db
       - rq
       - clamav
     volumes:
-      - static-volume:/app/static/
-      - media-volume:/app/media/
+      - static-volume:/opt/secure-record-transfer/static/
+      - media-volume:/opt/secure-record-transfer/media/
```

This directory doesn't have the static or media files anymore, it should be changed to:

```suggestion
      - static-volume:/opt/secure-record-transfer/app/static/
      - media-volume:/opt/secure-record-transfer/app/media/
```

You can leave the path in the NGINX container as-is, since it's fine for this volume to be mounted at a different path in that container. But this container does need the proper path since it has to be the path to the `STATIC_ROOT`.

---

Could you explain why it's fine for the static and media volumes to be mounted at a different path in the `nginx` container a bit further? Doesn't `nginx` need access to the files in `/opt/secure-record-transfer/app/static/` and `/opt/secure-record-transfer/app/media/` to be able to serve them?

---

Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/346/commits/a40bac9188e8a68d670122a302e4d1ccdc5d0887

---

> Could you explain why it's fine for the static and media volumes to be mounted at a different path in the `nginx` container a bit further? Doesn't `nginx` need access to the files in `/opt/secure-record-transfer/app/static/` and `/opt/secure-record-transfer/app/media/` to be able to serve them?

The filesystems for the `nginx` and `app` containers are completely separate, the only thing similar between them is the `static-volume` and `media-volume`. When `collecstatic` is run, the static files are all copied to the path `/opt/secure-record-transfer/app/static/`, which is in the `static-volume` volume. That volume could be mounted anywhere in the `nginx` container, so long as it knows where to find them.

So for example, this (partial) setup would be valid:

```yml
app:
  volumes:
    - static-volume:/opt/secure-record-transfer/app/static/
    - media-volume:/opt/secure-record-transfer/app/media/

nginx:
  volumes:
    - static-volume:/usr/share/nginx/html/static/
    - media-volume:/usr/share/nginx/html/media/
  environment:
    - STATIC_ROOT=/usr/share/nginx/html/static/
    - MEDIA_ROOT=/usr/share/nginx/html/media/
```

The static files are shared between the containers, but the path to them doesn't matter so much in the NGINX container.

---

Ah this makes sense. I forgot about how shared volumes between containers worked. Thanks for clarifying!